common/lib/xmodule/xmodule/modulestore/django.py

```diff
@@ -243,10 +244,43 @@ class ModuleI18nService(object):
     i18n service.
 
     """
-
     def __getattr__(self, name):
         return getattr(django.utils.translation, name)
 
+    def ugettext(self, string, **kwargs):
+        """
+        This operation is a proxy for django.utils.translation.ugettext, which is itself a proxy for
+        the GNU gettext ugettext operation.  Here we attempts to look up the provided string in the
+        XBlock's own domain translation catalog, currently expected to be:
+            <xblock_root>/conf/locale/<language>/LC_MESSAGES/<domain>.po
+        If ModuleI18nService can't locate the domain translation catalog then we fall-back onto
+        django.utils.translation.ugettext, which will attempt to find a matching string in the
+        LMS' own domain translation catalog -- effectively achieving translation by coincidence.
+        """
+        # The translation workflow should only execute if there's an actual string to look up
+        # If gettext processes an empty string the PO file header information will oddly be returned
+        translated_string = unicode(string)
+        if translated_string:
+            try:
+                # TODO: Move these values into settings.  We might allow for multiple possibilities
+                # for these values through the use of lists or tuples.  For example, instead of
+                # 'django' an XBlock developer could specify 'xblock_name' for their PO file.
+                xblock_domain = kwargs.get('xblock_domain', 'django')
+                xblock_locale = '/conf/locale'
+                locale_path = kwargs.get('xblock_root', '') + xblock_locale
+                selected_language = get_language()
+                translator = gettext.translation(
+                    xblock_domain,
+                    locale_path,
+                    [to_locale(selected_language if selected_language else settings.LANGUAGE_CODE)]
+                )
+                _ = translator.ugettext
+            except IOError:
+                _ = django.utils.translation.ugettext
+
+            translated_string = _(string)  # pylint: disable=translation-of-non-string
```

Comment:
@mattdrayer What is the purpose of passing `string` instead of `translated_string` here ?